clients/pkg/promtail/client/batch.go

```diff
@@ -72,19 +72,56 @@ func (b *batch) add(entry api.Entry) error {
 }
 
 func labelsMapToString(ls model.LabelSet, without ...model.LabelName) string {
+	var b strings.Builder
+	var labelsSize int
 	lstrs := make([]string, 0, len(ls))
+
 Outer:
 	for l, v := range ls {
 		for _, w := range without {
 			if l == w {
 				continue Outer
 			}
 		}
-		lstrs = append(lstrs, fmt.Sprintf("%s=%q", l, v))
+
+		str, s := labelNameValueToString(b, l, v)
+		lstrs = append(lstrs, str)
+		labelsSize += s
+	}
+	if len(lstrs) == 0 {
+		return "{}"
 	}
 
 	sort.Strings(lstrs)
-	return fmt.Sprintf("{%s}", strings.Join(lstrs, ", "))
+	return labelListToString(b, labelsSize, lstrs)
+}
+
+// labelListToString replicates 'fmt.Sprintf("{%s}", strings.Join(lstrs, ", "))', but in a more efficent way. We achieve this by using a string.Builder and keeping track of the size at an earlier stage
+func labelListToString(b strings.Builder, labelsSize int, lstrs []string) string {
+	b.Reset()
+	b.Grow(labelsSize + 2 + (2 * (len(lstrs) - 1)))
```

Comment:
Again, I'm not sure you need this. Did you measure without?

Reply:
This allows allocating the needed space from the beginning, instead of growing in smaller bits which causes allocations and copying of the underlying slice.